Control/DeepSeq/BackDoor.hs

```diff
@@ -5,7 +5,7 @@
 -- This module only re-export reasonably safe entities from non-safe
 -- modules when there is no safe alternative
 
-#if MIN_VERSION_base(4,9,0) || (MIN_VERSION_base(4,6,0) && !MIN_VERSION_base(4,7,0))
+#if (MIN_VERSION_base(4,9,0) && !MIN_VERSION_base(4,17,0)) || (MIN_VERSION_base(4,6,0) && !MIN_VERSION_base(4,7,0))
```

Comment:
Why `Data.ByteArray` is not Safe Haskell? Is this an oversight?

Reply:
The `Data.ByteArray` module exports indexing functions that segfault if you provide an out-of-bounds index.

Reply:
I’m kinda ambivalent here. My personal belief is that `Trustworthy` is suitable only for code, whose correctness was automatically verified. But if there is a strong desire to mark `Data.ByteArray` as such, I would not oppose. Note that there is not prior art: neither `Data.Primitive.ByteArray`, nor `Data.Text.Array`, nor `Data.ByteString.Short.Internal` are marked as safe, so I would not say that this breaks user expectations.

Reply:
`indexByteArray`

Reply:
@andrewthad none of the functions is exported:

```haskell
module Data.ByteArray (
  ByteArray(..)
) where
```